mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/MemoryDeletedMessagesVault.java

```diff
@@ -34,18 +40,25 @@
 
 public class MemoryDeletedMessagesVault implements DeletedMessageVault {
     private final Table<User, MessageId, DeletedMessage> table;
+    private final Table<User, MessageId, byte[]> contentTable;
```

Comment:
I'd rather put a Tuple in the previous table. It makes it simpler to handle atomicity